script.js

```diff
@@ -0,0 +1,368 @@
+  Enter: ['enter', '\n', '\n', '\n', '\n'],
+  ShiftLeft: ['shift'],
+  KeyZ: ['z', 'z', 'Z', 'я', 'Я'],
+  KeyX: ['x', 'x', 'X', 'ч', 'Ч'],
+  KeyC: ['c', 'c', 'C', 'с', 'С'],
+  KeyV: ['v', 'v', 'V', 'м', 'М'],
+  KeyB: ['b', 'b', 'B', 'и', 'И'],
+  KeyN: ['n', 'n', 'N', 'т', 'Т'],
+  KeyM: ['m', 'm', 'M', 'ь', 'Ь'],
+  Comma: ['comma', ',', '<', 'б', 'Б'],
+  Period: ['point', '.', '>', 'ю', 'Ю'],
+  Slash: ['questionMark', '/', '?', '.', ','],
+  ArrowUp: ['arrowUp', '▲', '▲', '▲', '▲'],
+  ShiftRight: ['rShift'],
+  ControlLeft: ['ctrl'],
+  MetaLeft: ['win'],
+  AltLeft: ['alt'],
+  Space: ['space', ' ', ' ', ' ', ' '],
+  AltRight: ['rAlt'],
+  ControlRight: ['rCtrl'],
+  ArrowLeft: ['arrowLeft', '◄', '◄', '◄', '◄'],
+  ArrowDown: ['arrowDown', '▼', '▼', '▼', '▼'],
+  ArrowRight: ['arrowRight', '►', '►', '►', '►'],
+};
+
+if (sessionStorage.lang === undefined) sessionStorage.lang = 'ENG';
+let lastPressButton = '';
+let shiftCase = false;
+let shiftCaseLeft = false;
+let altCaseLeft = false;
+let changeLangUpKeyFlag = false;
+let capsLock = false;
+
+function createVirtualKeyboard() {
+  const wrapper = document.createElement('div');
+  wrapper.className = 'wrapper';
+  document.body.append(wrapper);
+
+  const textarea = document.createElement('textarea');
+  textarea.className = 'textarea';
+  wrapper.append(textarea);
+
+  const keyboard = document.createElement('div');
+  keyboard.className = 'keyboard';
+  wrapper.append(keyboard);
+
+  const clearfix = document.createElement('div');
+  clearfix.className = 'clearfix';
+  keyboard.append(clearfix);
+
+  keyboard.after("Shift + alt - change language");
+}
+
+createVirtualKeyboard();
+
+const keyboard = document.querySelector('.keyboard');
+const textarea = document.querySelector('.textarea');
+const clearfix = document.querySelector('.clearfix');
+
+class Button {
+  constructor({ code, key, style }) {
+    this.className = `button ${style}`;
+    this.code = code;
+    this.key = key;
+  }
+
+  buildButton() {
+    const div = document.createElement('div');
+    div.className = this.className;
+    div.dataset.key = this.key;
+    div.dataset.code = this.code;
+    div.innerHTML = this.code;
+    return div;
+  }
+}
+
+function creatButton() {
+  const ar = [];
+  const arr = Object.keys(keyboardElements);
+  arr.forEach((el) => {
+    ar.push(
+      new Button({
+        code: el,
+        key: keyboardElements[el][0],
+        style: keyboardElements[el][0],
+      })
+    );
+  });
+  ar.forEach((el) => {
+    keyboard.append(el.buildButton());
+  });
+}
+
+creatButton();
+showKey();
+textarea.focus();
+
+function showKey() {
+  let arrValues = Object.values(keyboardElements);
+
+  arrValues.forEach((element) => {
+    if (element[0] === 'rShift') {
+      document.querySelector('.rShift').after(clearfix);
+    }
+  });
+
+  if (shiftCase === false && sessionStorage.lang === 'ENG') {
+    arrValues.forEach((element) => {
+      let a = document.querySelector(`.${element[0]}`);
```

Comment:
All method that works with browser internals are relatively expensive. As you can see from the image bellow that most of the time function just `query` element. You don't mutate DOM, so you can easily cache reference to every element.

<img width="880" alt="oleg virtual keyboard" src="https://user-images.githubusercontent.com/31409878/78799922-79f67500-79c3-11ea-8510-5ebd2a3fb305.png">